src/top-bar/period-select/period-menu.js

```diff
@@ -1,18 +1,32 @@
 import { Menu, MenuItem } from '@dhis2/ui'
+import moment from 'moment'
 import PropTypes from 'prop-types'
 import React from 'react'
+import { useAppContext } from '../../app-context/index.js'
 import { createHref } from '../../navigation/index.js'
 import { useSelectionContext } from '../../selection-context/index.js'
 import { getFixedPeriodsByTypeAndYear } from '../../shared/index.js'
 import classes from './period-menu.module.css'
 
+const formatPeriod = ({ period, periodType, dateFormat }) => {
+    if (periodType === 'Daily') {
+        return moment(period.displayName, 'YYYY-MM-DD').format(
+            // moment format tokens are case sensitive
+            // see https://momentjs.com/docs/#/parsing/string-format/
+            dateFormat === 'yyyy-mm-dd' ? 'YYYY-MM-DD' : 'DD-MM-YYYY'
```

Comment:
This assumes that there will only ever be two date formats, which might be brittle in the future.

Should this be extracted to a function?

Reply:
Yeah, I think your last suggestion would make most sense, i.e. to call `toUpperCase`.

Zooming out a little bit, I actually expected this functionality to be added to the fixed-periods module. That would involve a bit more work though:
- `getFixedPeriodTypes` would need to accept `formatYyyyMmDd` (and I guess `filterFuturePeriods`) too.
- Every function that calls `getFixedPeriodTypes` internally (i.e. `getFixedPeriodsByTypeAndYear`) would also need to accept these args.
- Then you would pass the formatter to `getFixedPeriodsByTypeAndYear` below